components/input-group/src/InputGroup.js

```diff
@@ -0,0 +1,99 @@
+/** @jsx jsx */
+
+import React, { Children, cloneElement } from 'react';
+import PropTypes from 'prop-types';
+import { jsx } from '@westpac/core';
+
+// ==============================
+// Component
+// ==============================
+
+/**
+ * Input Group: Styled input fields with attached addons. Addons can be text ($, %, .00 etc) or form controls (buttons or select inputs).
+ */
+export const InputGroup = ({ size, appearance, children, ...props }) => {
+	const childrenWithProps = Children.map(children, child =>
+		cloneElement(child, { size, appearance })
```

Comment:
I think size would be worthwhile, but appearance should be set per button. See comment in PropTypes below.

Thing is that size is also passed to the components via form, consumed from FormContext (assuming all form elements in a form should be the same size). So it's more likely they will receive a size value this way.

```
<Form size="large">
   <InputGroup>
      <TextInput />
      <Button>Go</Button>
   </InputGroup
</Form>
```